build fails for libastal-cava

**Describe the bug**
Build fails on Arch
```
[1/5] Compiling C object libastal-cava.so.0.1.0.p/cava.c.o
FAILED: libastal-cava.so.0.1.0.p/cava.c.o
cc -Ilibastal-cava.so.0.1.0.p -I. -I.. -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/cava -I/usr/include/cava/input -I/usr/include/cava/output -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -std=gnu11 -Wno-pedantic -Wno-unused-parameter -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/home/ferenyl/.cache/paru/clone/libastal-cava-git/src=/usr/src/debug/libastal-cava-git -flto=auto -fPIC -pthread -MD -MQ libastal-cava.so.0.1.0.p/cava.c.o -MF libastal-cava.so.0.1.0.p/cava.c.o.d -o libastal-cava.so.0.1.0.p/cava.c.o -c ../cava.c
../cava.c: In function ‘astal_cava_cava_start’:
../cava.c:254:69: error: passing argument 4 of ‘audio_raw_init’ from incompatible pointer type [-Wincompatible-pointer-types]
  254 |     audio_raw_init(&priv->audio_data, &priv->audio_raw, &priv->cfg, &priv->plan);
      |                                                                     ^~~~~~~~~~~
      |                                                                     |
      |                                                                     struct cava_plan *
In file included from /usr/include/cava/common.h:6,
                 from ../cava.c:1:
/usr/include/cava/output/common.h:49:39: note: expected ‘struct cava_plan **’ but argument is of type ‘struct cava_plan *’
   49 |                    struct cava_plan **plan);
      |                    ~~~~~~~~~~~~~~~~~~~^~~~
ninja: build stopped: subcommand failed.
==> ERROR: A failure occurred in build().
    Aborting...
error: failed to build 'libastal-cava-git-r617.12fa0fb-2':
error: packages failed to build: libastal-cava-git-r617.12fa0fb-2
```

**To Reproduce**
install via aur or download repo and run 
```
meson setup build
meson install -C build
```

**Expected behavior**
Build should not end in error

**Additional context**
Add any other context about the problem here.

--- a/lib/cava/cava.c
+++ b/lib/cava/cava.c
@@ -2,6 +2,7 @@
 #include <gio/gio.h>
 
 #include "astal-cava.h"
+#include "cava/common.h"
 #include "cava/config.h"
 #include "glib-object.h"
 #include "glib.h"
@@ -27,7 +28,7 @@ struct _AstalCavaCava {
 };
 
 typedef struct {
-    struct cava_plan * plan;
+    struct cava_plan* plan;
     struct config_params cfg;
     struct audio_data audio_data;
     struct audio_raw audio_raw;
@@ -100,9 +101,14 @@ static void astal_cava_cava_cleanup(AstalCavaCava* self) {
 
     cava_destroy(priv->plan);
 
+    audio_raw_clean(&priv->audio_raw);
+
     g_free(priv->audio_data.cava_in);
     g_free(priv->audio_data.source);
 
+    // use config_clean(&priv->cfg); instead.
+    // While this was patched into the libcava 0.10.6 AUR package, it is not included in the
+    // official release, so wait for the next tagged release.
     g_free(priv->cfg.audio_source);
     g_free(priv->cfg.raw_target);
     g_free(priv->cfg.data_format);

--- a/lib/cava/cava.c
+++ b/lib/cava/cava.c
@@ -27,7 +27,7 @@ struct _AstalCavaCava {
 };
 
 typedef struct {
-    struct cava_plan plan;
+    struct cava_plan * plan;
     struct config_params cfg;
     struct audio_data audio_data;
     struct audio_raw audio_raw;
@@ -77,7 +77,7 @@ static gboolean exec_cava(AstalCavaCava* self) {
 
     pthread_mutex_lock(&priv->audio_data.lock);
     cava_execute(priv->audio_data.cava_in, priv->audio_data.samples_counter,
-                 priv->audio_raw.cava_out, &priv->plan);
+                 priv->audio_raw.cava_out, priv->plan);
     if (priv->audio_data.samples_counter > 0) priv->audio_data.samples_counter = 0;
     pthread_mutex_unlock(&priv->audio_data.lock);
 
@@ -98,7 +98,7 @@ static void astal_cava_cava_cleanup(AstalCavaCava* self) {
     pthread_mutex_unlock(&priv->audio_data.lock);
     g_thread_join(priv->input_thread);
 
-    cava_destroy(&priv->plan);
+    cava_destroy(priv->plan);
 
     g_free(priv->audio_data.cava_in);
     g_free(priv->audio_data.source);

--- a/lib/cava/meson.build
+++ b/lib/cava/meson.build
@@ -22,7 +22,7 @@ install_headers('astal-cava.h')
 
 cava = dependency(
   'cava',
-  version: '>=0.10.3',
+  version: '>=0.10.6',
   required: true,
   fallback: ['cava', 'cava_dep'],
 )

--- a/lib/cava/subprojects/cava.wrap
+++ b/lib/cava/subprojects/cava.wrap
@@ -1,7 +1,7 @@
 [wrap-file]
-directory = cava-0.10.4
-source_url = https://github.com/LukashonakV/cava/archive/0.10.4.tar.gz
-source_filename = cava-0.10.4.tar.gz
-source_hash = 7bc1c1f9535f2bcc5cd2ae8a2434a2e3a05f5670b1c96316df304137ffc65756
+directory = cava-0.10.6
+source_url = https://github.com/LukashonakV/cava/archive/0.10.6.tar.gz
+source_filename = cava-0.10.6.tar.gz
+source_hash = e715c4c6a625b8dc063e57e8e81c80e4d1015ec1b98db69a283b2c6770f839f4
 [provide]
 cava = cava_dep

